kb/development/mbedtls-coding-standards.md

```diff
@@ -116,11 +116,15 @@ All public names (functions, variables, types, `enum` constants, macros) must st
     mbedtls_aes_setkey_decrypt()
 ```
 
+Note that all externally linked functions must have a name starting with `mbedtls_` to avoid link-time conflicts, even if they are not declared in a public header. This also applies to global variables (which should be used very sparingly).
+
 Exception: code implementing the PSA crypto API uses the `PSA_` and `psa_` prefixes. This includes official APIs as well as draft APIs that are on the PSA standards track. API extensions which are meant to remain specific to Mbed TLS, and internal functions, should use the `MBEDTLS_/mbedtls_` prefix, however there are many existing cases of using the `PSA_/psa_` prefix.
 
 ### Local names
 
-Static functions and macros that are not in public headers follow the same convention except the initial `MBEDTLS_` or `mbedtls_` prefix: they start directly with the function name.
+Static functions do not need to have the `mbedtls_` prefix. They usually start with the module name (which makes it slightly easier to follow call stacks across modules), but this is not compulsory.
+
+Macros defined only in the `library` directory should have the `MBEDTLS_` prefix. It's ok to use shorter names if it's convenient, but avoid names without underscores as some embedded platforms define short macro names in their system headers. Exception: macros defined in headers used by alternative implementations or PSA drivers (including headers that they include) must start with with `MBEDTLS_` or `PSA_`.
```

Comment:
I am not sure I understand. Instead of
> Macros defined only in the `library` directory should have the `MBEDTLS_` prefix.
Did you mean
> Macros defined only in the `library` directory don't need to have the `MBEDTLS_` prefix.

Reply:
I did mean to recommend sticking to the `MBEDTLS_` prefix. We've had problems before with our short names clashing with platform headers that also defined short names (not C compliant, but we had to cope).

Reply:
Ah, I see. Then by

> It's ok to use shorter names if it's convenient

you mean

> It's ok to omit the `MBEDTLS_` prefix if it is convenient

Right?

Reply:
Yes. And that it's also ok to omit the module name (and we do have a few macros already that don't have a module name because they make sense across modules). But keep it sensible and don't define something that's likely to clash like `ASSERT`.

Reply:
Sounds good, could you please clarify this as well when you are doing the update?